controllers/slicegateway/slicegateway.go

```diff
@@ -1375,6 +1376,21 @@ func (r *SliceGwReconciler) ReconcileGatewayDeployments(ctx context.Context, sli
 		}
 	}
 
+	// Create PodDisruptionBudget for slice gateway's pod to at least have 1 instance of pods on each worker
+	// when disruption has occurred.
+	//
+	// Note: This should run an attempt to create PDB regardless of whether current reconciliation creating deployments
+	// as the request could've been requeued due to failure at the creation of PDB.
+	if err = r.createPodDisruptionBudgetForSliceGatewayPods(ctx, log, r.Client, sliceName, sliceGw); err != nil {
+		log.Error(err, "Failed to create PodDisruptionBudget for SliceGW deployments",
+			"SliceName", sliceName, "SliceGwName", sliceGwName)
+
+		return ctrl.Result{
+			// Requeue for reconciler to try creating PDB in subsequent runs
+			Requeue: true,
+		}, nil, true
```

Comment:
```suggestion
		return ctrl.Result{}, err, true
```
Let's return error back, returning errors always requeues the reconciler.

Reply:
Noted. Updated it. Thanks!